File: bubbleSort.py
a = [5,4,3,2,1]
n = len(a)
for i in range(n):

    for j in range(0,n-i-1):
        print("n-i-1 ::: ", n-i-1)
        if arr[j] > arr[j+1] :
            a[j], a[j+1] = a[j+1], a[j]
    
print(a)

File: bubbleSort.js
const sort = (arr) => {
  for (let i = 0; i < arr.length; i++) {
    let swapped = false
    for (let j = 1; j < arr.length-i; j++) {
      if (arr[j] < arr[j-1]) {
        // s w a p
        let temp = arr[j]
        arr[j] = arr[j-1]
        arr[j-1] = temp
        swapped = true
      }
    }
    if (!swapped) break;
  }
  return arr
}

const main = () => {
  const arr = [1,2,3,4,5]
  const sortedArr = sort(arr)
  console.log("sortedArr :: ", sortedArr)
}

main()
